Homework3

alexfromsao opened the pull request.

alexfromsao committed:
Книга Властелин колец

Сделана разметка книги.
Подключен похожий шрифт и Google fonts.

alexfromsao committed:
Поправлены отступы в коде

Изменил отступы, поменял язык страницы, перенёс ссылку в main

Ambiens commented on book.html:
```diff
@@ -0,0 +1,63 @@
+        <link rel="preconnect" href="https://fonts.googleapis.com">
+        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+        <link href="https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,400;0,800;1,400&display=swap" rel="stylesheet">
+        <link rel="stylesheet" href="book/book-style.css">
```
Путь не правильный. Видимо не проверяли даже.

alexfromsao replied:
++
поправил
произошло из-за того, что не разобрался с гитом и изначально хотел сделать это задание отдельно, а не в общей ветке.

Ambiens commented on book.html:
```diff
@@ -0,0 +1,63 @@
+        <title>Текст книги Властелин Колец</title>
+    </head>
+    <body>
+        <header>
```
Структура странное. Всё можно в main вложить.
А далее оперируем статьями. Тут нет layout чтобы делить на header и т.п., ни какой-то навигации. Это текст книги.

alexfromsao replied:
++

Ambiens commented on book.html:
```diff
@@ -0,0 +1,63 @@
+            <h1>Автор, Дж.Р.Р.</h1>
+            <p>
+                Толкин, профессор Оксфордского университета, специалист по древнему и средневековому английскому языку, создал удивительный мир - Средиземье, который вот уже без малого пятьдесят лет неодолимо влечет к себе миллионы читателей.
+            </p>
```
Заголовок и рассказ об авторе. Это блок можно выделить как отдельную статью.
Главный заголовок тут это Книга Пятая. Так что Автор, Дж.Р.Р. это уже подзаголовок.

alexfromsao replied:
++

Ambiens commented on book.html:
```diff
@@ -0,0 +1,63 @@
+            </nav>
+        </header>
+
+        <main>        
```
Не понятно, почему main тут оборачивает книгу. Статья об авторе к книге не относится? Содержание тоже не относится?
Вся книга вместе с заголовком это article

alexfromsao replied:
не совсем понял, что требуется сделать

Ambiens commented on book.html:
```diff
@@ -0,0 +1,63 @@
+            <a href="#chapter1">Вернуться к главе 1</a>
+        </main>   
+        
+    </body>
```
Комментарии в коде, оформляем отдельной строкой, на том же уровне, что и блок к которому относится

alexfromsao replied:
++

Ambiens reviewed (commented):
Добавил замечания

alexfromsao commented on book.html:
```diff
@@ -0,0 +1,63 @@
+                    <li>Глава 2 Выбор Арагорна</li>
+                    <li>Глава 3 Рохан собирает войска</li>
+                </ul>
+            </nav>
```
++

alexfromsao commented on book.html:
```diff
@@ -0,0 +1,63 @@
+                    <span class="bold">Яркий желтый огонь</span> в темном небе заставил Пиппина съежиться от страха. Зачем Гэндальф везет его в это страшное, там, впереди? Он протер глаза. На востоке всходила луна. Значит, они будут скакать и скакать, мчаться сквозь ночь бесконечно долго.
+                </p>
+
+            </section>
```
++

alexfromsao commented on book.html:
```diff
@@ -0,0 +1,63 @@
+                <h2 id="chapter3">Глава 3 Рохан собирает войска</h2>
+                <p> <!-- Ему было интересно в незнакомой стране курсив --> 
+                    Мерри с изумлением озирался по сторонам. <span class="italic">Ему было интересно в незнакомой стране.</span> Это был край без неба; в туманной дымке глаз различал только горы, каменная стена вставала за стеной, меж ними разверзались пропасти, заполненные туманом. От могучих водопадов воздух тихо звенел, шумели деревья, стучали по камням копыта. <span class="underline">Раньше М ...
+                </p> <!-- Раньше Мерри любил слушать о горах, думать о них, но теперь, когда горные хребты вздымались вокруг под самые небеса, он чувствовал себя неуютно, живо ощущая исполинскую тяжесть, гнетущую грудь Средиземья. подчёркнутый --> 
```
++
ссылка на css была битая, поправил

alexfromsao committed:
Исправления на основе комментариев

Ambiens commented on book.html:
```diff
@@ -0,0 +1,66 @@
+                    <!-- Раньше Мерри любил слушать о горах, думать о них, но теперь, когда горные хребты вздымались вокруг под самые небеса, он чувствовал себя неуютно, живо ощущая исполинскую тяжесть, гнетущую грудь Средиземья. подчёркнутый -->
+                    Мерри с изумлением озирался по сторонам. <i>Ему было интересно в незнакомой стране.</i> Это был край без неба; в туманной дымке глаз различал только горы, каменная стена вставала за стеной, меж ними разверзались пропасти, заполненные туманом. От могучих водопадов воздух тихо звенел, шумели деревья, стучали по камням копыта. <u>Раньше Мерри любил слушать о горах, думать о них,  ...
+                </p>  
+            </article>
```
Вся книга, это тоже один большой article, без нижний ссылки.
И вся книга со статьёй об авторе это тоже ещё больший article, без нижний ссылки.

Я бы добавил такие две дополнительные вложенности.
Т.е. main  > article с текстом + ссылка. В article с текстом > article об авторе + article c книгой. article c книгой > глава + глава + глава

Ambiens commented on book.html:
```diff
@@ -0,0 +1,66 @@
+            </ul>
+
+            <article>
+                <h2 id="chapter1">Глава 1 Минас Тирит</h2>
```
Уже писал, внимательнее. id="chapter1" должен стоять на разделе, а не на заголовках и чём-то ещё.
И ссылаемся мы на статью, а не на заголовок статьи. А если перед заголовком будет отступ или иллюстрация, мы её проедим мимо?

alexfromsao replied:
Учёл на будущее
++

Ambiens reviewed (commented):
Уже почти идеально.
Но! Типографики текста, так и не вижу. Не уважаете вы читателя. (

alexfromsao committed:
Внесение правок в соответствии с коментариями

alexfromsao committed:
Удалена вложенность и оттипографлен текст

Удалены лишние теги
текс прогнан через типограф
Увеличено расстояние между строк
Сcылки на глав перенесены в article
